### PR TITLE
document how to use at_test_logs to check for no warnings

### PR DESCRIPTION
**Pull request by ranocha**
This documents how to use `@test_logs` to check for no warnings. Additionally, I added `Logging.` to the `min_level`s to make it more clear (since `min_level=Warn` does not work on Julia v1.5, even after `using Logging`).

### CODE REVIEW

**Comment by ranocha** [comment]
I don't see how the changes of this PR could cause the test failure.

**Comment by ranocha** [comment]
An appropriate label would be `doc` (but I can't add labels).

**Comment by ranocha** [comment]
Bump. Is there any interest in this?

**Comment by vtjnash** [comment]
It looks like something similar was added in #39670, but I thought the extra details here were useful. I've taken the liberty of combining them and refining the text some more for tone.

File: stdlib/Test/src/logging.jl [hunk 1]
```diff
@@ -135,21 +135,26 @@ We can test the info message using
...
-    @test_logs (:info,) (:debug,"Iteration 42") min_level=Debug match_mode=:any foo(100)
+    @test_logs (:info,) (:debug,"Iteration 42") min_level=Logging.Debug match_mode=:any foo(100)
 
 The macro may be chained with `@test` to also test the returned value:
 
     @test (@test_logs (:info,"Doing foo with n=2") foo(2)) == 42
 
-If you want to test an absence of logger messages, you can pass no log_patterns:
+If you want to test for the absence of warnings, you can omit specifying log
+patterns to [`@test_nowarn`](@ref) and set the `min_level` accordingly:
```
Comment (vtjnash) [inline stdlib/Test/src/logging.jl:150]:
```suggestion
patterns and set the `min_level` accordingly:
```

**Pull request merged**